### ISSUE TITLE
False positive on select {}

### ISSUE THREAD

**Issue by abhinav** [mgechev/revive#804]
**Describe the bug**

**To Reproduce**
1. I updated revive `go install github.com/mgechev/revive@latest`
2. I run it with the following flags & configuration file:

```shell
$ cat > main.go << EOF
import "net/http"

func main () {
  go http.ListenAndServe()
  select {}
}
EOF

$ revive -set_exit_status ./...
```

**Expected behavior**
No errors.

**Logs**

```
empty-block: this block is empty, you can remove it (revive)
```

**Desktop (please complete the following information):**
 - OS: Linux
 - Version of Go: 1.20.2

**Additional context**
Previously reported in #698

**Comment by chavacava** [comment]
Hi @abhinav, thanks for filling the issue (and providing a PR)
(IMO this is not a _false positive_, the block is actually empty.)

As I've mentioned in the clone of this issue (#698) the frequency of the construction `select {}` is low. Moreover, this construction blocks control forever thus warning on it seems to me a good idea.

**Comment by abhinav** [comment]
Hey, @chavacava. Thanks for the response and for maintaining the tool.

RE: false positive:
I disagree that this is not a false positive.
Yes, the block is empty, but there's a material difference in behavior between the block being there and not.
Sorry, let me clarify: when the linter complains about the empty block in `if foo {}`, it makes sense because deleting the block is equivalent behavior. No-op to no-op.
However, in the case of `select {}`, deleting the block does not turn a no-op to no-op. It turns a line that blocks the goroutine into a no-op, which will then allow the goroutine to exit.
Note, for example, that the rule also [ignores empty functions](https://github.com/mgechev/revive/blob/5db07b5c971648c8c59d0c84d176e94f11dfbcec/rule/empty-block.go#L38) because deleting an empty method body can have a material difference on the program's behavior (e.g. if it was implementing an interface).

I agree that as of #698, the frequency of `select {}` was an adequate reason not to add this.
However, I've added legitimate use cases of `select {}` in my PR.
Given that this rule is now in the default rule set, avoiding false positives where possible is desirable.
